test/e2e/utils_test.go

```diff
@@ -90,6 +92,97 @@ func checkIfAllMemberClustersHaveJoined() {
 	}
 }
 
+// createResourcesForFleetGuardRail create resources required for guard rail E2Es.
+func createResourcesForFleetGuardRail() {
+	cr := rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster-role",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"*"},
+				Verbs:     []string{"*"},
+				Resources: []string{"*"},
+			},
+		},
+	}
+	Eventually(func() error {
+		return hubClient.Create(ctx, &cr)
+	}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "failed to create cluster role %s for fleet guard rail E2E", cr.Name)
+
+	crb := rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster-role-binding",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "User",
+				Name:     "test-user",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     "test-cluster-role",
+		},
+	}
+
+	Eventually(func() error {
+		return hubClient.Create(ctx, &crb)
+	}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "failed to create cluster role binding %s for fleet guard rail E2E", crb.Name)
+}
+
+// deleteResourcesForFleetGuardRail deletes resources created for guard rail E2Es.
+func deleteResourcesForFleetGuardRail() {
+	crb := rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster-role-binding",
+		},
+	}
+	Expect(hubClient.Delete(ctx, &crb)).Should(Succeed())
+
+	cr := rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-cluster-role",
+		},
+	}
+	Expect(hubClient.Delete(ctx, &cr)).Should(Succeed())
+}
+
+func createMemberClusterResource(name string) {
+	// Create the MC.
+	By(fmt.Sprintf("In MC BeforeAll %s", name))
+	mc := &clusterv1beta1.MemberCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: clusterv1beta1.MemberClusterSpec{
+			Identity: rbacv1.Subject{
+				Name:      testUser,
+				Kind:      "ServiceAccount",
+				Namespace: utils.FleetSystemNamespace,
+			},
+			HeartbeatPeriodSeconds: 60,
+		},
+	}
+	Expect(hubClient.Create(ctx, mc)).To(Succeed(), "Failed to create MC %s", mc)
+}
+
+func deleteMemberClusterResource(name string) {
+	By(fmt.Sprintf("In MC AfterAll %s", name))
+	Eventually(func(g Gomega) error {
+		var mc clusterv1beta1.MemberCluster
+		err := hubClient.Get(ctx, types.NamespacedName{Name: name}, &mc)
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		g.Expect(err).Should(Succeed(), "Failed to get MC %s", name)
+		g.Expect(hubClient.Delete(ctx, &mc)).Should(Succeed())
```

Comment:
Hi Arvind! A nit: for this one it might be better to check for actual removal after sending the delete request.